src/eddymotion/model/base.py

```diff
@@ -210,27 +246,25 @@ def predict(self, gradient, **kwargs):
         return retval
 
 
-class TrivialB0Model:
+class TrivialB0Model(BaseDWIModel):
     """A trivial model that returns a *b=0* map always."""
 
-    __slots__ = ("_S0",)
-
-    def __init__(self, S0=None, **kwargs):
+    def __init__(self, **kwargs):
         """Implement object initialization."""
-        if S0 is None:
-            raise ValueError("S0 must be provided")
+        super().__init__(**kwargs)
 
-        self._S0 = S0
+        if self._S0 is None:
+            raise ValueError("S0 must be provided")
 
-    def fit(self, *args, **kwargs):
+    def fit(self, data, **kwargs):
         """Do nothing."""
 
-    def predict(self, gradient, **kwargs):
+    def predict(self, *_, **kwargs):
         """Return the *b=0* map."""
         return self._S0
 
 
-class AverageDWModel:
```

Comment:
Why was the `I` from `DWI` droped for `AverageDWModel` (vs `AverageDWIModel` (or vs `dMRI`, `DMRI`).

This dwi vs dmri naming discrepancy (if it is such) is also present in other place (and other repositories, e.g. `nipreps`). At some point a decision will need to be made to make things consistent (or the policy be explained if there is one).

Reply:
Can be addressed at a later time.  For now, the relevant thing is to have a proper base model to inherit from. Resolving.